src/Middleware/OutputCaching/src/OutputCacheEntry.cs

```diff
@@ -69,26 +66,9 @@ public bool TryFindHeader(string key, out StringValues values)
     internal void SetBody(ReadOnlySequence<byte> value, bool recycleBuffers)
     {
         Body = value;
-        _recycleBuffers = recycleBuffers;
-    }
-
-    public void Dispose()
-    {
-        var headers = Headers;
-        var body = Body;
-        Headers = default;
-        Body = default;
-        Recycle(headers);
-        RecyclableReadOnlySequenceSegment.RecycleChain(body, _recycleBuffers);
-        // ^^ note that this only recycles the chain, not the actual buffers
-    }
-
-    private static void Recycle<T>(ReadOnlyMemory<T> value)
-    {
-        if (MemoryMarshal.TryGetArray<T>(value, out var segment) && segment.Array is { Length: > 0 })
-        {
-            ArrayPool<T>.Shared.Return(segment.Array);
-        }
+        _ = recycleBuffers; // satisfy IDE0060
```

Comment:
We set `recycleBuffers` to true in one case
https://github.com/dotnet/aspnetcore/blob/9922772d07a7f4d21d2becdb00fb5a206b9807cc/src/Middleware/OutputCaching/src/OutputCacheMiddleware.cs#L431
Are we now adding unnecessary strain on the ArrayPool by not returning that memory?

Reply:
array pool doesn't care about non-returned buffers - they'll just GC

Reply:
Sure, but if you drain the array pool it becomes expensive to get an array from the pool. We saw this recently with the IdentityModel package where they weren't disposing an object that had rented memory from the array pool and so it was GCed instead of returned, but then the array pool became empty and it was very expensive to rent new arrays from the pool.

Reply:
@BrennanConroy right; but if we don't want to leak buffers, we need to either a: implement ref-counting, or b: fetch the memory from somewhere else, and drop *that* on the floor; I had a stab at the ref-counting originally, and ironically: it was that damned missing `throw;` that caused that experiment to fail; we can implement it; consider: https://github.com/dotnet/aspnetcore/pull/51131/files

but... that's what I was saying is a bit more "fun"; is it worth deferring that to service release 1 ?

Reply:
am I understanding correctly though that this takes us back to .NET 7? if so, then it's apparently shippable quality. or is this a place where we will be introducing something new.

Reply:
> am I understanding correctly though that this takes us back to .NET 7? 

The PR as is doesn't take us all the way back to .NET 7. In .NET 7 we just allocated. This PR reverts to an in-between state, we call `ArrayPool.Rent` but don't call `ArrayPool.Return`.

I think we should either
* run a quick perf test to see if it causes issues 
* just remove ArrayPool.Rent and allocate
* Fix ArrayPool usage (there is a draft PR by Marc for this)

Reply:
your judgement, but I'd err towards minimum risk given we have no more previews. .NET 7 was successful.